hugr-core/src/hugr/linking.rs

```diff
@@ -0,0 +1,520 @@
+//! Directives and errors relating to linking Hugrs.
+
+use std::{collections::HashMap, fmt::Display};
+
+use itertools::Either;
+
+use crate::{
+    Hugr, HugrView, Node,
+    core::HugrNode,
+    hugr::{HugrMut, internal::HugrMutInternals},
+};
+
+/// Methods for linking Hugrs, i.e. merging the Hugrs and adding edges between old and inserted nodes.
+///
+/// This is done by module-children from the inserted (source) Hugr replacing, or being replaced by,
+/// module-children already in the target Hugr; static edges from the replaced node,
+/// are transferred to come from the replacing node, and the replaced node(/subtree) then deleted.
+pub trait LinkHugr: HugrMut {
+    /// Copy nodes from another Hugr into this one, with linking directives specified by Node.
+    ///
+    /// If `parent` is non-None, then `other`'s entrypoint-subtree is copied under it.
+    /// `children` of the Module root of `other` may also be inserted with their
+    /// subtrees or linked according to their [NodeLinkingDirective].
+    ///
+    /// # Errors
+    ///
+    /// * If `children` are not `children` of the root of `other`
+    /// * If `parent` is Some, and `other.entrypoint()` is either
+    ///   * among `children`, or
+    ///   * descends from an element of `children` with [NodeLinkingDirective::Add]
+    ///
+    /// # Panics
+    ///
+    /// If `parent` is `Some` but not in the graph.
+    #[allow(clippy::type_complexity)]
+    fn insert_from_view_link_nodes<H: HugrView>(
```

Comment:
These names make sense, but I wonder if we could shorten them a bit -- or do you plan on renaming to `insert_hugr` and `insert_from_view` once those are deleted?

I can suggest:
- `insert_link_from_view`
- `insert_link_hugr`

I think they are just as clear and sound like natural "evolutions" of the current API. Up to you :)